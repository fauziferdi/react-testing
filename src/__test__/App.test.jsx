import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../App";

describe("App component testing", () => {
  test("render component", () => {
    render(<App />);
    const viteLogo = screen.getByAltText("Vite logo");
    expect(viteLogo).toBeInTheDocument();
  });

  test("render react logo", () => {
    render(<App />);
    const reactLogo = screen.getByAltText("React logo");
    expect(reactLogo).toBeInTheDocument();
  });

  test("display initial count 0", () => {
    render(<App />);
    const countElement = screen.getByText(/count is 0/i);
    expect(countElement).toBeInTheDocument();
  });

  test("increment count", () => {
    render(<App />);
    const button = screen.getByRole("button");
    fireEvent.click(button);
    const countElement = screen.getByText(/count is 1/i);
    expect(countElement).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
