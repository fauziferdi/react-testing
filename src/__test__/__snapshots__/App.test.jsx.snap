// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component testing matches snapshot 1`] = `
<div>
  <div>
    <a
      href="https://vite.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="[object Object]"
      />
    </a>
    <a
      href="https://react.dev"
      target="_blank"
    >
      <img
        alt="React logo"
        class="logo react"
        src="[object Object]"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.jsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
